autogpt/processing/text.py

```diff
@@ -67,6 +68,7 @@ def summarize_text(
     """
     if not text:
         return "Error: No text to summarize"
+    MEMORY = get_memory_instance()
```

Comment:
Why move this here? (just curious)

Reply:
This is so that the memory instance is created within the context of this function, instead of the global context, that helps isolate all other functions so they can be tested separately.

Reply:
Memory is a singleton.

Reply:
@BillSchumacher lol, your comment undid the approval